notes/zrq/20220526-03-blue-deploy.txt

```diff
@@ -0,0 +1,1488 @@
+    >     },
+    >     {
+    >       "id": "2H3TBEN9A",
+    >       "path": "/Users/ciNg2thu/7. Good astrometric solutions via ML Random Forrest classifier"
+    >     },
+    >     {
+    >       "id": "2H4P4F7ED",
+    >       "path": "/Users/ciNg2thu/8. Tips and tricks"
+    >     }
+    >   ]
+
+
+# -----------------------------------------------------
+# Run all the test user's notebooks.
+#[root@ansibler]
+
+    for noteid in $(
+        curl \
+            --silent \
+            --cookie "${zepcookies:?}" \
+            "${zeppelinurl:?}/api/notebook" \
+        | jq -r ".body[] | select(.path | startswith(\"/Users/${testusername:?}\")) | .id"
+        )
+    do
+        echo "-----------------------------------------------------"
+        echo "Notebook [${noteid}]"
+        echo
+        zepnbclear     "${noteid}"
+        echo
+        zepnbexecstep  "${noteid}"
+        echo
+        zepnbstatus    "${noteid}"
+        echo
+        zepnbtotaltime "${noteid}"
+        echo
+    done
+
+    >   -----------------------------------------------------
+    >   Notebook [2H5NA8EWW]
+    >   ....
+    >   ....
+    >   Para [20210510-105824_1166659944][Example code from previous cell]
+    >   {
+    >     "status": "OK",
+    >     "body": {
+    >       "code": "ERROR"
+    >     }
+    >   }
+    >   Result [ERROR]
+    >   ....
+    >   ....
+    >   -----------------------------------------------------
+    >   Notebook [2H4Z656G8]
+    >   ....
+    >   ....
+    >   Para [20210507-124436_282498395][Database and table details]
+    >   {
+    >     "status": "OK",
+    >     "body": {
+    >       "code": "ERROR"
+    >     }
+    >   }
+    >   Result [ERROR]
+    >   ....
+    >   ....
+    >   
+    >       #
+    >       # Still got lots of errors.
+    >       #
+    >   
+    >   # -----------------------------------------------------
+    >   # -----------------------------------------------------
+    >   # Fix the errors on Zeppelin.
+    >   #[user@zeppelin]
+    >   
+    >       pushd zeppelin/logs
+    >   
+    >           less zeppelin-interpreter-spark-ciNg2thu-ciNg2thu-fedora-iris-gaia-blue-20220526-zeppelin.log
+    >   
+    >       popd
+    >   
+    >   ....
+    >    INFO [2022-05-27 04:44:15,869] ({FIFOScheduler-interpreter_2088462041-Worker-1} Logging.scala[logInfo]:57) - Preparing resources for our AM container
+    >   ERROR [2022-05-27 04:44:15,870] ({FIFOScheduler-interpreter_2088462041-Worker-1} Logging.scala[logError]:94) - Error initializing SparkContext.
+    >   java.lang.IllegalArgumentException: java.net.UnknownHostException: albert
+    >           at org.apache.hadoop.security.SecurityUtil.buildTokenService(SecurityUtil.java:447)
+    >           at org.apache.hadoop.hdfs.NameNodeProxiesClient.createProxyWithClientProtocol(NameNodeProxiesClient.java:131)
+    >   ....
+
+
+    sudo vi /opt/hadoop/etc/hadoop/hdfs-site.xml
```

Comment:
Should this be applied to our deployment scripts? And is there a reason to go for "albert" rather than the default "user"?

Reply:
Nevermind, I think you did include it. Let me debug what went wrong with my deploy

Reply:
Apologies I thought I fixed the extra '/', my bad.

Reply:
'albert' is just for this iteration, something deliberately not in the existing source code.

I chose 'albert' to aid debugging because it is guaranteed not to be in anything ales and so is easy to track down in the logs and configuration files using `grep`. We will probably want to revert back to `user`, but there is something to discuss before we do (I'll add a new issue to discuss it).